go/cmd/coordinator/cmd.go

```diff
@@ -62,6 +62,11 @@ func init() {
 	// Compaction service Memberlist
 	Cmd.Flags().StringVar(&conf.CompactionServiceMemberlistName, "compaction-memberlist-name", "compaction-service-memberlist", "Compaction memberlist name")
 	Cmd.Flags().StringVar(&conf.CompactionServicePodLabel, "compaction-pod-label", "compaction-service", "Compaction pod label")
+
+	// Block store provider
+	Cmd.Flags().StringVar(&conf.BlockStoreProvider, "block-store-provider", "none", "Block store provider")
```

Comment:
discussed offline: there might be some invalid combinations here for e.g. if `block-store-provider` is none but `version-file-enabled` is true. Add checks to disallow such combinations?

Related question is why can't you get rid of `version-file-enabled` and only have three possible values of `block-store-provider`: none to denote that versioning is disabled, minio for minio and s3 for s3?